addons/web/static/src/views/relational_model.js

```diff
@@ -1762,18 +1762,14 @@ class DynamicList extends DataPoint {
         const lastIndex = Math.max(fromIndex, toIndex) + 1;
         let reorderAll = records.some((record) => record.data[handleField] === undefined);
         if (!reorderAll) {
-            let lastSequence = (asc ? -1 : 1) * Infinity;
-            for (let index = 0; index < records.length; index++) {
-                const sequence = getSequence(records[index]);
-                if (
-                    ((index < firstIndex || index >= lastIndex) &&
-                        ((asc && lastSequence >= sequence) ||
-                            (!asc && lastSequence <= sequence))) ||
-                    (index >= firstIndex && index < lastIndex && lastSequence === sequence)
-                ) {
+            // if the list is not already ordered by sequence (strict), reorder all
+            const sequences = records.map(getSequence);
+            for (let i = 0; i < sequences.length - 1; i++) {
+                let [seq1, seq2] = sequences.slice(i, i + 2);
```

Comment:
I personally find this more readable
```suggestion
                let seq1 = sequences[i];
                let seq2 = sequences[i + 1];
```

Reply:
To be clear, do you want to go forward with my original proposal? @hubvd seemed really against it for good reasons.

Reply:
I mean, it fixes the reported issue. As we can't really merge the orm bug in stable, I'm not against it. We shouldn't do this in master though